wallet/src/actors/worker/methods.rs

```diff
@@ -703,16 +692,9 @@ impl Worker {
             log::debug!("[SU] Received chain: {:?}", block_chain);
 
             // For each of the blocks we have been informed about, ask a Witnet node for its contents
-            for ChainEntry(epoch, id) in block_chain {
+            for ChainEntry(_epoch, id) in block_chain {
                 let get_block_future = self.get_block(id.clone());
-                let block: types::ChainBlock = futures03::executor::block_on(get_block_future)?;
-
-                // Compute if block should be considered confirmed
-                // Note: blocks confirmed in past superblocks are considered confirmed
-                // FIXME(#1437): best approach would be the node signaling if blocks are confirmed
-                let node_tip_superblock_index = self.current_epoch()? / superblock_period;
-                let local_tip_superblock_index = epoch / superblock_period;
-                let confirmed = node_tip_superblock_index - local_tip_superblock_index > 1;
+                let (block, confirmed) = futures03::executor::block_on(get_block_future)?;
```

Comment:
Won't this degrade the performance of the synchronization progress significantly? If we get the blocks list in batches, maybe we could also ask for their confirmation status in batches. Or maybe offer that information upfront in the blocks list.

Reply:
I don't know about performance, but yes, we could expose a `getLastConfirmedBlockEpoch` method in the JSON-RPC interface, and set all blocks before than to confirmed.

Reply:
So I guess that we can settle with this PR for now and do what you just mentioned in a subsequent update, when reducing synchronization times start to be more pressing as the chain grows.